src/libraries/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/SharedPerformanceCounter.cs

```diff
@@ -1691,6 +1691,7 @@ private unsafe void Initialize(string fileMappingName, int fileMappingSize, int
                         throw new InvalidOperationException(SR.SetSecurityDescriptorFailed);
 
                     Interop.Kernel32.SECURITY_ATTRIBUTES securityAttributes = default;
+                    securityAttributes.lpSecurityDescriptor = securityDescriptorPointer.DangerousGetHandle();
```

Comment:
Is it possible to use the SafeHandle here and avoid calling DangerousGetHandle?

Reply:
I see, we're using the shared definition which defines this as IntPtr.  It looks like we're ensuring the SafeHandle stays alive the entire method, so this should be ok.  Just better to avoid unsafe if we can.